packages/app/src/app/pages/Dashboard/Content/routes/RepositoryBranches/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { useParams } from 'react-router-dom';
+import { useAppState, useActions } from 'app/overmind';
+import { Header } from 'app/pages/Dashboard/Components/Header';
+import { VariableGrid } from 'app/pages/Dashboard/Components/VariableGrid';
+import { DashboardGridItem, PageTypes } from 'app/pages/Dashboard/types';
+import { SelectionProvider } from 'app/pages/Dashboard/Components/Selection';
+import { Element } from '@codesandbox/components';
+import { useWorkspaceSubscription } from 'app/hooks/useWorkspaceSubscription';
+import { PrivateRepoFreeTeam } from 'app/pages/Dashboard/Components/Repository/stripes';
+import { getProjectUniqueKey } from 'app/overmind/namespaces/dashboard/utils';
+
+export const RepositoryBranchesPage = () => {
+  const params = useParams<{ path: string }>();
+  const path = params.path || '';
+  const [, owner, name] = path.split('/');
+  const actions = useActions();
+  const {
+    activeTeam,
+    dashboard: { repositoriesWithBranches, viewMode },
+  } = useAppState();
+
+  const key = getProjectUniqueKey({ teamId: activeTeam, owner, name });
+  const repositoryProject = repositoriesWithBranches[key] || undefined;
+
+  React.useEffect(() => {
+    // If no repositories were fetched yet and the user tries
+    // to directly access a repository, we should fetch said
+    // repository only.
+    if (!repositoryProject && activeTeam) {
+      actions.dashboard.getRepositoryByWithBranches({
+        activeTeam,
+        owner,
+        name,
+      });
+    }
+  }, [path, activeTeam]);
```

Comment:
Do we need to specify `path` here? Since it's a page, it should re-render every time the page is changed, calling this effect anyway? Also, when changing a team we direct the user to the `/recent` page right? If all these are the case, we can move this out of the effect completely I reckon!

Reply:
Right, this might be leftover from the original page I didn't consider